types/gdal/gdal-tests.ts

```diff
@@ -0,0 +1,78 @@
+import * as gdal from 'gdal';
+
+// gdal.checksumImage(srcRasterBand, 0, 0, 512, 512);
+
+gdal.config.get('GDAL_DATA');
+gdal.config.set('GDAL_DATA', 'C:\\warmerda\\bld\\data');
+
+/*gdal.contourGenerate({
+	src: srcRasterBand,
+	dst: dstRasterBand
+});*/
+
+gdal.decToDMS(60, 'lat');
+gdal.decToDMS(60, 'lat', 2);
+
+/*gdal.fillNodata({
+
+});*/
+
+let creationOptionsArray: string[];
+creationOptionsArray = ['VRT_CREATION_OPTION=...'];
+let creationOptionsObject: object;
+creationOptionsObject = {
+	VRT_CREATION_OPTION: '...'
+};
+gdal.open('C:\\datasets\\ogr.shp');
+gdal.open('C:\\datasets\\ogr.shp', 'r');
+gdal.open('C:\\datasets\\ogr.shp', 'r', 'ESRI Shapefile');
+gdal.open('C:\\datasets\\ogr.shp', 'r', ['ESRI Shapefile']);
+gdal.open('C:\\datasets\\ogr.shp', 'r+');
+gdal.open('C:\\datasets\\ogr.shp', 'r+', 'ESRI Shapefile');
+gdal.open('C:\\datasets\\ogr.shp', 'r+', ['ESRI Shapefile']);
+gdal.open('C:\\datasets\\raster.vrt', 'w', 'VRT');
+gdal.open('C:\\datasets\\raster.vrt', 'w', 'VRT', 512, 512, 3, 1, creationOptionsArray);
+gdal.open('C:\\datasets\\raster.vrt', 'w', 'VRT', 512, 512, 3, 1, creationOptionsObject);
+gdal.open('C:\\datasets\\raster.vrt', 'w', ['VRT']);
+gdal.open('C:\\datasets\\raster.vrt', 'w', ['VRT'], 512, 512, 3, 1, creationOptionsArray);
+gdal.open('C:\\datasets\\raster.vrt', 'w', ['VRT'], 512, 512, 3, 1, creationOptionsObject);
+
+/*let polygonizeOptions: gdal.PolygonizeOptions;
```

Comment:
Yes!
These are some examples of how to use it
I've commented them because gdal is using datasets read from files, so we cannot test it, but i've tested it out and works very well.